src/Parse/ParseQuery.php

```diff
@@ -416,6 +416,65 @@ public function count($useMasterKey = false)
     }
 
     /**
+     * Execute a distinct query and return unique values.
+     *
+     * @param string $key field to find distinct values
+     *
+     * @return array
+     */
+    public function distinct($key)
+    {
+        $sessionToken = null;
+        if (ParseUser::getCurrentUser()) {
```

Comment:
You can do something like this to avoid calling `ParseUser::getCurrentUser()` twice, while still validating there is a current user.
```php
if($user = ParseUser::getCurrentUser()) {
  ...
}
```
`$user` would be set in the conditional block then to use.